snuba/datasets/metrics_aggregate_processor.py

```diff
@@ -0,0 +1,166 @@
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import Any, Mapping, Optional, Sequence, Tuple
+
+from snuba.consumers.types import KafkaMessageMetadata
+from snuba.processor import (
+    AggregateInsertBatch,
+    MessageProcessor,
+    ProcessedMessage,
+    _ensure_valid_date,
+)
+from snuba.query.expressions import (
+    Expression,
+    FunctionCall,
+    Literal,
+    OptionalScalarType,
+)
+
+
+def _literal(value: OptionalScalarType) -> Literal:
+    return Literal(None, value)
+
+
+def _array_literal(values: Sequence[OptionalScalarType]) -> FunctionCall:
+    return FunctionCall(None, "array", tuple(map(_literal, values)))
+
+
+def _call(function_name: str, arguments: Tuple[Expression, ...]) -> FunctionCall:
+    return FunctionCall(None, function_name, arguments)
+
+
+METRICS_SET_TYPE = "s"
+METRICS_DISTRIBUTIONS_TYPE = "d"
+METRICS_COUNTERS_TYPE = "c"
+
+
+class MetricsAggregateProcessor(MessageProcessor, ABC):
+    TEN_SECONDS = 10
+    ONE_HOUR = 3600
+    ONE_DAY = 3600 * 24
+    GRANULARITIES_SECONDS = [TEN_SECONDS, ONE_HOUR, ONE_DAY]
```

Comment:
Pretty sure we have the 1 minute granularity as well.
https://github.com/getsentry/snuba/blob/master/snuba/migrations/snuba_migrations/metrics/templates.py#L243

Reply:
when I run migrations in my local I see:
```
7b701b64404c :) show tables;

SHOW TABLES

┌─name──────────────────────────────────┐
│ discover_local                        │
│ errors_local                          │
│ groupassignee_local                   │
│ groupedmessage_local                  │
│ metrics_buckets_local                 │
│ metrics_counters_buckets_local        │
│ metrics_counters_local                │
│ metrics_counters_mv_10s_local         │
│ metrics_counters_mv_3600s_local       │
│ metrics_counters_mv_86400s_local      │
│ metrics_counters_mv_local             │
│ metrics_distributions_buckets_local   │
│ metrics_distributions_local           │
│ metrics_distributions_mv_10s_local    │
│ metrics_distributions_mv_3600s_local  │
│ metrics_distributions_mv_86400s_local │
│ metrics_distributions_mv_local        │
│ metrics_sets_local                    │
│ metrics_sets_mv_10s_local             │
│ metrics_sets_mv_3600s_local           │
│ metrics_sets_mv_86400s_local          │
│ metrics_sets_mv_local                 │
│ migrations_local                      │
│ outcomes_hourly_local                 │
│ outcomes_mv_hourly_local              │
│ outcomes_raw_local                    │
│ sentry_local                          │
│ sessions_hourly_local                 │
│ sessions_hourly_mv_local              │
│ sessions_raw_local                    │
│ transactions_local                    │
└───────────────────────────────────────┘

31 rows in set. Elapsed: 0.011 sec. 
```

and on prod from the admin tool:
```
{
  "column_names": [
    "name"
  ],
  "rows": [
    [
      "metrics_buckets_dist"
    ],
    [
      "metrics_buckets_local"
    ],
    [
      "metrics_counters_buckets_dist"
    ],
    [
      "metrics_counters_buckets_local"
    ],
    [
      "metrics_counters_dist"
    ],
    [
      "metrics_counters_local"
    ],
    [
      "metrics_counters_mv_10s_local"
    ],
    [
      "metrics_counters_mv_3600s_local"
    ],
    [
      "metrics_counters_mv_86400s_local"
    ],
    [
      "metrics_counters_mv_local"
    ],
    [
      "metrics_distributions_buckets_dist"
    ],
    [
      "metrics_distributions_buckets_local"
    ],
    [
      "metrics_distributions_dist"
    ],
    [
      "metrics_distributions_local"
    ],
    [
      "metrics_distributions_mv_10s_local"
    ],
    [
      "metrics_distributions_mv_3600s_local"
    ],
    [
      "metrics_distributions_mv_86400s_local"
    ],
    [
      "metrics_distributions_mv_local"
    ],
    [
      "metrics_sets_dist"
    ],
    [
      "metrics_sets_local"
    ],
    [
      "metrics_sets_mv_10s_local"
    ],
    [
      "metrics_sets_mv_3600s_local"
    ],
    [
      "metrics_sets_mv_86400s_local"
    ],
    [
      "metrics_sets_mv_local"
    ],
    [
      "asynchronous_metrics"
    ],
    [
      "metrics"
    ]
  ],
  "input_query": "select name from system.tables where name like '%metrics%' (metrics_counters,snuba-metrics-0-1:9000)"
}

```

Reply:
```
CREATE MATERIALIZED VIEW default.metrics_sets_mv_local TO default.metrics_sets_local
(
    `org_id` UInt64,
    `project_id` UInt64,
    `metric_id` UInt64,
    `granularity` UInt32,
    `tags.key` Array(UInt64),
    `tags.value` Array(UInt64),
    `timestamp` DateTime,
    `retention_days` UInt16,
    `value` AggregateFunction(uniqCombined64, UInt64)
) AS
SELECT
    org_id,
    project_id,
    metric_id,
    60 AS granularity,
    tags.key,
    tags.value,
    toStartOfInterval(timestamp, toIntervalSecond(60)) AS timestamp,
    retention_days,
    uniqCombined64State(arrayJoin(set_values)) AS value
FROM default.metrics_buckets_local
WHERE materialization_version = 0
GROUP BY
    org_id,
    project_id,
    metric_id,
    tags.key,
    tags.value,
    timestamp,
    granularity,
    retention_days │
```

metrics_sets_mv_local (the default ones are 60 seconds aggregations)

Reply:
got it, I'll add that in